modules/data-entry/src/main/frontend/src/questionnaireEditor/DroppableAnswerOption.jsx

```diff
@@ -0,0 +1,225 @@
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useEffect, useRef, useState } from "react";
+import { createPortal } from 'react-dom';
+import PropTypes from "prop-types";
+import {
+  Checkbox,
+  Grid,
+  IconButton,
+  TextField,
+  Tooltip,
+} from "@mui/material";
+import CloseIcon from '@mui/icons-material/Close';
+import DragIndicatorIcon from '@mui/icons-material/DragIndicator';
+
+import { draggable, dropTargetForElements, } from '@atlaskit/pragmatic-drag-and-drop/element/adapter';
+import { setCustomNativeDragPreview } from '@atlaskit/pragmatic-drag-and-drop/element/set-custom-native-drag-preview';
+import { pointerOutsideOfPreview } from '@atlaskit/pragmatic-drag-and-drop/element/pointer-outside-of-preview';
+import { combine } from '@atlaskit/pragmatic-drag-and-drop/combine';
+import { attachClosestEdge, extractClosestEdge, } from '@atlaskit/pragmatic-drag-and-drop-hitbox/closest-edge';
+import { DropIndicator } from "@atlaskit/pragmatic-drag-and-drop-react-drop-indicator/box";
+import invariant from 'tiny-invariant';
+
+
+const valueKey = Symbol("value"); // creates a unique property keys
+
+function getOptionData(value) {
+  return { [valueKey]: true, valueId: value.value };
+}
+
+export function isOptionData(data) {
+  return data[valueKey] === true;
+}
+
+function DroppableAnswerOption(props) {
+  const { classes, value, index, deleteOption, generateDescriptionIcon } = props;
+
+  const ref = useRef(null);
+
+  /* Object to maintain the draggable state
+    type DraggableState =
+      | { type: "idle" }
+      | { type: "preview"; container: HTMLElement }
+      | { type: "dragging" }
+      | { type: 'dragging-over'; closestEdge: Edge | null }
+  */
+  const [draggableState, setDraggableState] = useState({ type: 'idle' });
+
+  useEffect(() => {
+    const element = ref.current;
+    invariant(element);
+
+    const data = getOptionData(value);
+
+    return combine(
+      draggable({
+        element,
+        getInitialData() {
+            return data;
+        },
+        onGenerateDragPreview({ nativeSetDragImage }) {
+            setCustomNativeDragPreview({
+                nativeSetDragImage,
+                getOffset: pointerOutsideOfPreview({
+                    x: '4px',
+                    y: '4px',
+                }),
+                render({ container }) {
+                    setDraggableState({ type: 'preview', container });
+                },
+            });
+        },
+        onDragStart() {
+            setDraggableState({ type: "dragging" });
+        },
+        onDrop() {
+            setDraggableState({ type: 'idle' });
+        },
+    }), 
+    dropTargetForElements({
```

Comment:
The indentation seems off here.
```suggestion
      }),
      dropTargetForElements({
```